src/worker-thread/canvas/WebGLRenderingContextPolyfill.ts

```diff
@@ -0,0 +1,2051 @@
+    this.MAX_SAMPLES,
+    this.MAX_3D_TEXTURE_SIZE,
+    this.MAX_ARRAY_TEXTURE_LAYERS,
+    this.MAX_TEXTURE_LOD_BIAS,
+    this.MAX_UNIFORM_BUFFER_BINDINGS,
+    this.MAX_UNIFORM_BLOCK_SIZE,
+    this.UNIFORM_BUFFER_OFFSET_ALIGNMENT,
+    this.MAX_COMBINED_UNIFORM_BLOCKS,
+    this.MAX_COMBINED_VERTEX_UNIFORM_COMPONENTS,
+    this.MAX_COMBINED_FRAGMENT_UNIFORM_COMPONENTS,
+    this.MAX_TRANSFORM_FEEDBACK_INTERLEAVED_COMPONENTS,
+    this.MAX_TRANSFORM_FEEDBACK_SEPARATE_ATTRIBS,
+    this.MAX_TRANSFORM_FEEDBACK_SEPARATE_COMPONENTS,
+    this.MAX_ELEMENT_INDEX,
+    this.MAX_SERVER_WAIT_TIMEOUT,
+  ];
+
+  private readonly _serializedAsTransferrableObject: number[];
+
+  private _parameters: { [key: number]: any };
+  private _supportedExtensions: string[] | null = null;
+  private _extensions: { [key: string]: any };
+  private _contextAttributes: WebGLContextAttributes | null = null;
+  private readonly _buffers: { [key: string]: vGLBuffer | null } = {
+    arrayBuffer: null,
+    elementArrayBuffer: null,
+    copyReadBuffer: null,
+    copyWriteBuffer: null,
+    transformFeedbackBuffer: null,
+    uniformBuffer: null,
+    pixelPackBuffer: null,
+    pixelUnpackBuffer: null,
+    framebuffer: null,
+    renderbuffer: null,
+    drawFramebuffer: null,
+    readFramebuffer: null,
+  };
+  private readonly _bindings: { [key: string]: TransferrableGLObject | null } = {
+    texture2D: null,
+    program: null,
+    vertexArray: null,
+    transformFeedback: null,
+    sampler: null,
+  };
+  // https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/isEnabled
+  // By default, all capabilities except gl.DITHER are disabled.
+  private readonly _capabilities: { [key: GLenum]: boolean } = {
+    [this.DITHER]: true,
+    [this.BLEND]: false,
+    [this.CULL_FACE]: false,
+    [this.DEPTH_TEST]: false,
+    [this.POLYGON_OFFSET_FILL]: false,
+    [this.SAMPLE_ALPHA_TO_COVERAGE]: false,
+    [this.SAMPLE_COVERAGE]: false,
+    [this.SCISSOR_TEST]: false,
+    [this.STENCIL_TEST]: false,
+    [this.RASTERIZER_DISCARD]: false,
+  };
+
+  constructor(id: number, canvas: HTMLCanvasElement, contextAttributes?: WebGLContextAttributes | null) {
+    super();
+    this.id = id;
+    this._serializedAsTransferrableObject = [TransferrableObjectType.TransferObject, this.id];
+    this.canvas = canvas;
+
+    this.drawingBufferHeight = canvas.height;
+    this.drawingBufferWidth = canvas.width;
+
+    this._parameters = {};
+    this._extensions = [];
+    this._contextAttributes = contextAttributes || null;
+
+    callFunction(this.canvas.ownerDocument as Document, this, 'getContextAttributes', [])
+      .then((result) => (this._contextAttributes = result))
+      .catch((reason) => {
+        console.warn('Failed to get WebGL context attributes', reason);
+      });
+
+    callFunction(this.canvas.ownerDocument as Document, this, 'getSupportedExtensions', [])
+      .then((result) => (this._supportedExtensions = result))
```

Comment:
may be unassigned if someone in worker would ask it before this initialization

Reply:
fixed